job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PostBuildScriptsContext.groovy

```diff
@@ -28,4 +30,18 @@ class PostBuildScriptsContext implements Context {
     void onlyIfBuildSucceeds(boolean onlyIfBuildSucceeds = true) {
         this.onlyIfBuildSucceeds = onlyIfBuildSucceeds
     }
+
+    /**
+     * If set, runs the build steps only if the build was failure. Defaults to {@code false}.
+     */
+    void onlyIfBuildFails(boolean onlyIfBuildFails = true) {
```

Comment:
Add a `@RequiresPlugin` annotation with a minimum version to all new methods to indicate that the methods needs a certain plugin version.